docs/modules/nifi/pages/usage_guide/custom-components/custom-python-processors.adoc

```diff
@@ -0,0 +1,148 @@
+          - name: nifi-python-extensions
+            configMap:
+              name: nifi-python-extensions
+          - name: nifi-python-working-directory
+            emptyDir: {}
+----
+
+[#git-sync]
+=== 2. Use git-sync
+
+As an alternative you can use `git-sync` to keep your Python processors up to date.
+You need to add a sidecar using podOverrides that syncs into a shared volume between the `nifi` and `git-sync` container.
+
+The following snippet can serve as a starting point (the Git repo has the folder `processors` with the Python scripts inside).
+
+[source,yaml]
+----
+spec:
+  nodes:
+    podOverrides:
+      spec:
+        containers:
+          - name: nifi
+            volumeMounts:
+              - name: nifi-python-extensions
+                mountPath: /nifi-python-extensions
+              - name: nifi-python-working-directory
+                mountPath: /nifi-python-working-directory
+          - name: git-sync
+            image: registry.k8s.io/git-sync/git-sync:v4.2.3
+            args:
+              - --repo=https://github.com/stackabletech/nifi-talk
+              - --root=/nifi-python-extensions
+              - --period=10s
+            volumeMounts:
+              - name: nifi-python-extensions
+                mountPath: /nifi-python-extensions
+        volumes:
+          - name: nifi-python-extensions
+            emptyDir: {}
+          - name: nifi-python-working-directory
+            emptyDir: {}
+----
+
+Afterwards you need to update your source directory (you added previously) accordingly to point into the Git subfolder you have.
+
+[source,yaml]
+----
+spec:
+  nodes:
+    configOverrides:
+      nifi.properties:
+        # Replace the property from the previous step
+        # Format is /nifi-python-extensions/<git-repo-name>/<git-folder>/
+        nifi.python.extensions.source.directory.custom: /nifi-python-extensions/nifi-talk/processors/
+----
+
+=== 3. Use PersistentVolume
+
+You can also mount a PVC below `/nifi-python-extensions` using podOverrides and shell into the NiFi Pod to make changes.
+However, the <<configmap>> or <<git-sync>> approach is recommended.
+
+== Check processors have been loaded
+
+NiFi logs every Python processor it found.
+You can use that to check if the processors have been loaded.
+
+[source,console]
+----
+$ kubectl logs nifi-2-0-0-node-default-0 -c nifi | grep -P 'Discovered Python Processor|Discovered or updated [0-9]+ Python Processors'
+2025-02-14 14:40:20,694 INFO [main] o.a.n.n.StandardExtensionDiscoveringManager Discovered Python Processor PythonZgrepProcessor
+2025-02-14 14:40:20,697 INFO [main] o.a.n.n.StandardExtensionDiscoveringManager Discovered Python Processor TransformOpenskyStates
+2025-02-14 14:40:20,700 INFO [main] o.a.n.n.StandardExtensionDiscoveringManager Discovered Python Processor UpdateAttributeFileLookup
+2025-02-14 14:40:20,700 INFO [main] o.a.n.n.StandardExtensionDiscoveringManager Discovered or updated 3 Python Processors in 60 millis
```

Comment:
Reformatted (and tested) with a maximum line length of 75:

```suggestion
$ kubectl logs nifi-2-0-0-node-default-0 -c nifi \
      | grep 'Discovered.*Python Processor'
… INFO [main] …  Discovered Python Processor PythonZgrepProcessor
… INFO [main] …  Discovered Python Processor TransformOpenskyStates
… INFO [main] …  Discovered Python Processor UpdateAttributeFileLookup
… INFO [main] …  Discovered or updated 3 Python Processors in 64 millis
```

Reply:
Regarding the line-break before grep: I personally prefer one-lines but that's personal taste.
Thanks for shortening the logs!